### PR TITLE
Update CI

### PR DESCRIPTION
**Pull request by achaikou**

### CODE REVIEW

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    branches: [master]
+  workflow_dispatch:
+
+jobs:
+  windows_build:
+    runs-on: windows-2019
+    strategy:
+      matrix:
+        platform:
+        - x86
```
Comment (yngve793) [inline .github/workflows/windows.yml:15]: Does this means that it tries both x86(32 bits) and x64?

**Review by yngve793 — APPROVED** [review]
All actions works. I do not know if there is a simple way to test the final product.

Read through the yml files and nothing stands out. 

I presume that the catch.hpp file is cut and pase as to generated.

**Review by ajaust — APPROVED** [review]
Looks good to me overall. I left some comments about potential renamings and some questions because I did not understand everything immediately. You are allowed to ignore my comments. 😄 

Nice job with the modularisation of the workflows. :) 

I added the repository to Snyk for extra scanning (and compliance).

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,334 +0,0 @@
-version: 2.1
-
-orbs:
```
[the remark this answers is not in the document]
Comment (achaikou) [reply .circleci/config.yml:3]: Good catch!
File remade into .md and updated.
Workflow link I hope will start working after merge.

**Review by achaikou — COMMENTED** [review]
Thanks for the review, comments addressed!

Yep, for catch2 I just changed the file with the downloaded version.

Workflow import into github actions was done with gthub own tool. It did a pretty good job, split mostly everything  correctly into the workflow files, so good job to the tool :smile:

File: .github/workflows/osx.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    branches: [master]
+  workflow_dispatch:
+
+jobs:
+  macos_build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [macos-11]
```
[the remark this answers is not in the document]
Comment (ajaust) [reply .github/workflows/osx.yml]: I agree that one test probably enough. I am not an expert either.

File: .github/workflows/arch.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+name: Architectures
```
[the remark this answers is not in the document]
Comment (ajaust) [reply .github/workflows/arch.yml:1]: Awesome, thanks for the clarification. I agree that testing dlisio on different architecture is definitely what we should do. I guess I misread the action.

**Review by ajaust — APPROVED** [review]
Looks good to me. 👍

**Comment by yngve793** [comment]
No further comments.

**Pull request merged**